lib/ansible/modules/cloud/vmware/vcenter_folder.py

```diff
@@ -230,25 +279,31 @@ def ensure(self):
                                               " exception %s " % to_native(gen_exec))
             self.module.exit_json(**results)
 
-    def get_folder(self, folder_name, folder_type, parent_folder=None):
+    def get_folder(self, datacenter_name, folder_name, folder_type, parent_folder=None):
         """
         Get managed object of folder by name
         Returns: Managed object of folder by name
 
         """
         folder_objs = get_all_objs(self.content, [vim.Folder], parent_folder)
         for folder in folder_objs:
-            if folder.name == folder_name and \
-               self.datacenter_folder_type[folder_type].childType == folder.childType:
-                return folder
+            if parent_folder:
+                if folder.name == folder_name and \
+                   self.datacenter_folder_type[folder_type].childType == folder.childType:
+                    return folder
+            else:
+                if folder.name == folder_name and \
+                   self.datacenter_folder_type[folder_type].childType == folder.childType and \
+                   folder.parent.parent.name == datacenter_name:    # e.g. folder.parent.parent.name == /DC01/host/folder
```

Comment:
I am not sure about this condition. I will test few scenarios and will comment back.

Reply:
It just checks if the parent (datacenter) of the parent (e.g. `host`) is the datacenter. E.g. `folder.parent.parent.name == /DC01/host/folder`. I tested it several times with two different datacenters and host/vm folders.